app/helpers/textual_summary_helper.rb

```diff
@@ -56,7 +56,7 @@ def textual_key_value_group(items)
   def textual_tags
     label = _("%{name} Tags") % {:name => session[:customer_name]}
     h = {:label => label}
-    tags = session[:assigned_filters]
+    tags = textual_record_tags
```

Comment:
I don't think you should touch this line, it is a global setting for all our textual summaries. I think you should search for the `session[:assigned_filters]`, where its value for the Datastores is being assigned and how it differs from other places like VMs.

Reply:
Even though using the `session` is a wrong pattern here...this is not the place for refactoring.

Reply:
I understand, please compare how the feeding of the session value for the right and wrong implementations is different and change them right at the source.

Reply:
The difference is that the feeding of the session value does not happen at all, for wrong implementations.